addons/website_sale_stock/static/src/js/variant_mixin.js

```diff
@@ -61,6 +62,28 @@ VariantMixin._onChangeCombinationStock = function (ev, $parent, combination) {
         }
     }
 
+    // needed xml-side for formatting of remaining qty
+    combination.formatQuantity = (qty) => {
+        if (Number.isInteger(qty)) {
+            return qty;
```

Comment:
`free_qty` is a Float field. Why do you need `isInteger`?

Reply:
By default, the `uom_rounding` for (e.g.) 'Units' is set to 0.01.
But up to now, if the number was always an integer (regardless of actual type) it would be displayed as such (since it wasn't formatted), and I think that's the desired behavior?
Maybe we could have got away without `isInteger` if we use a method that drops trailing zeros, but if in the end we base it off the `uom_rounding` I think it makes more sense to always have the same number of digits, and there would still be the problem of checking this case and manually removing the trailing decimal separator.